appyx-components/demos/common/src/commonMain/kotlin/com/bumble/appyx/components/demos/puzzle15/ui/Puzzle15MotionController.kt

```diff
@@ -0,0 +1,82 @@
+package com.bumble.appyx.components.demos.puzzle15.ui
+
+import DefaultAnimationSpec
+import androidx.compose.animation.core.SpringSpec
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.DpOffset
+import androidx.compose.ui.unit.dp
+import com.bumble.appyx.components.demos.puzzle15.Puzzle15Model
+import com.bumble.appyx.components.demos.puzzle15.Puzzle15Model.Tile
+import com.bumble.appyx.components.demos.puzzle15.operation.Swap
+import com.bumble.appyx.interactions.core.ui.context.UiContext
+import com.bumble.appyx.interactions.core.ui.gesture.Drag
+import com.bumble.appyx.interactions.core.ui.gesture.Gesture
+import com.bumble.appyx.interactions.core.ui.gesture.GestureFactory
+import com.bumble.appyx.interactions.core.ui.gesture.dragDirection4
+import com.bumble.appyx.interactions.core.ui.property.impl.Position
+import com.bumble.appyx.interactions.core.ui.state.MatchedTargetUiState
+import com.bumble.appyx.transitionmodel.BaseMotionController
+
+class Puzzle15MotionController(
+    uiContext: UiContext,
+    defaultAnimationSpec: SpringSpec<Float> = DefaultAnimationSpec
+) : BaseMotionController<Tile, Puzzle15Model.State, MutableUiState, TargetUiState>(
+    uiContext = uiContext,
+    defaultAnimationSpec = defaultAnimationSpec
+) {
+
+    override fun Puzzle15Model.State.toUiTargets(): List<MatchedTargetUiState<Tile, TargetUiState>> =
+        items.mapIndexed { index, tileElements ->
+            MatchedTargetUiState(
+                element = tileElements, // todo: if i create items here this misbehaves because the ids will be different
+                targetUiState = TargetUiState(
+                    position = Position.Target(
+                        value = DpOffset(
+                            x = (index % 4 * 60).dp,
```

Comment:
I wonder if instead baking in `60`, we could take the total width/height from transition bounds, and just divide them by 4. This would make the entire layout flexible.